include/common.fypp

```diff
@@ -74,6 +74,12 @@
 $:"s" if cmplx=="c" else "d" if cmplx=="z" else "x" if cmplx=="y" else "q" if cmplx=="w" else "ERROR"
 #:enddef
 
+#! BLAS/LAPACK/Linear Algebra Integer Kinds
+#:set LINALG_INT_KINDS       = ["ilp","ilp64"]
```

Comment:
Using fypp it would be possible then to initialize the 3 lists with only the 32bit support and then add a conditional something like
```
#:set LINALG_INT_KINDS       = ["ilp"]
#:set LINALG_INT_SUFFIX      = [""]
#:if WITH_LINALG_ILP64
#:set LINALG_INT_KINDS       = LINALG_INT_KINDS+["ilp64"]
#:set LINALG_INT_SUFFIX      = LINALG_INT_SUFFIX+["_I64"]
#:endif
#:set LINALG_INT_TYPES       = ["integer({})".format(k) for k in LINALG_INT_KINDS]
#:set LINALG_INT_KINDS_TYPES = list(zip(LINALG_INT_KINDS, LINALG_INT_TYPES, LINALG_INT_SUFFIX))
```
Like this, it would be easy to parametrize which version to preprocess in case one doesn't require the ilp64 version. This can be added then as a parameter in the fypp_deployment.py script:
```python
parser.add_argument("--with_linalg_ilp64",action='store_true', help="Include the WITH_LINALG_ILP64 preprocessing keyword for BLAS/LAPACK")
```

Reply:
Amazing, thanks @jalvesz. See the deployment config edits in [34b7a81](https://github.com/fortran-lang/stdlib/pull/888/commits/34b7a81ad657789f6beaeff8af657dd181eae7b1).
I've used `WITH_ILP64` instead of `WITH_LINALG_ILP64` because it's shorter and seems closer to the other flags `WITH_QP` etc. What do you think? Also `WITH_LINALG_ILP64` is OK imho.

Reply:
I like also the shorter keyword! Perfect!

I'm not sure regarding the default, I'm guessing it should be important to weight: binary size, compilation time and users demand for ilp64 support out of the box. No dogma on my side.

Reply:
Default `ILP64` build is currently `False`.